### PR TITLE
Devp/js action

### PR DESCRIPTION
**Pull request by markpitchless**
Add JS action to run harness deploy.

Tested and working: https://github.com/fac/freeagent/compare/dgholz/sandbox_deploy_action

Draft:
* [x] History rewrite, collapsing
* [x] Work out what to do with the example

Fixes fac/dev-platform#270

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,16 +6,59 @@ Call the Harness.io API to deploy an application.
...
+   - name: Deploy Link
+     run: |
+       echo harness_url: ${{ steps.deploy.outputs.harness_url }}
+```
+
 ## Inputs
 
+### webhookUrl
+
+The full webhook url for harness, including the authentication token. You **should** pass this as a secret.
```
Comment (dgholz) [inline README.md:28]: drop the should, 'Add the URL as an Actions secret on your repo, and pass it to the action'

File: .github/workflows/example.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Example Harness Deploy
```
Comment (dgholz) [inline .github/workflows/example.yml:1]: delete, move to the readme

Comment (stevenwilliamson) [reply .github/workflows/example.yml:1]: 👍 move to Readme, so it's in the docs front and center when you hit the repo, or change filename so it's ignored harness-deploy.yml.example for example :)

File: README.md [hunk 3]
```diff
@@ -6,16 +6,59 @@ Call the Harness.io API to deploy an application.
 
 ## Usage
 
+```yaml
+   - name: Deploy
+     id: deploy
+     uses: fac/harness-deploy-action@v1
+     with:
+       webhookUrl: ${{ secrets.HARNESS_WEBHOOK_URL }}
+       application: XYZ12345678ABC
```
Comment (stevenwilliamson) [inline README.md:15]: Add a snippet explaining to users where to find this values ?

File: README.md [hunk 4]
```diff
@@ -6,16 +6,59 @@ Call the Harness.io API to deploy an application.
...
+### version
+
+String version to deploy.
+
 ## Outputs
 
-## Secrets
+### harness_url
+
+The harness url to watch the deploy on. Note that this contains the application
```
Comment (stevenwilliamson) [inline README.md:50]: note that it should be kept secret?

**Review by stevenwilliamson — COMMENTED** [review]
Few tiny comments no blockers.

When merged down it would be useful to add a dependabot config to keep the node_modules up to date and avoid bit rot / tech-debt.

File: README.md [hunk 5]
```diff
@@ -6,16 +6,59 @@ Call the Harness.io API to deploy an application.
...
+       echo harness_url: ${{ steps.deploy.outputs.harness_url }}
+```
+
 ## Inputs
 
+### webhookUrl
+
+The full webhook url for harness, including the authentication token. You **should** pass this as a secret.
+
+e.g. `https://app.harness.io/gateway/api/webhooks/XXXXXXXX?accountId=YYYYYYYYYY`, where `XXXXXXXX` and `YYYYYYYY` are actual tokens. You can get this from the harness UI.
```
Comment (dgholz) [inline README.md:30]: I feel like describing the format of the URL isn't as useful as explaining how to set up a webhook trigger with a custom payload & copying the manual link

the Harness docs are… not very clear on how to do it: https://docs.harness.io/article/ys3cvwm5gc-trigger-a-deployment-on-git-event#step_1_add_a_trigger
Use a 'Custom' Payload type

File: action.yml [hunk 6]
```diff
@@ -3,15 +3,15 @@ name: Harness Deploy Action
 author: FreeAgent
 description: Call the Harness.io API to deploy an application
 inputs:
+  webhookUrl:
+    description: The secret
```
Comment (dgholz) [inline action.yml:7]:
```suggestion
    description: The Manual URL of your Harness Trigger
```

File: .github/workflows/example.yml (hunk 2 above)
Comment (markpitchless) [reply .github/workflows/example.yml:1]: It is, I already added the Usage docs. Will remove the example.

File: README.md (hunk 4 above)
Comment (markpitchless) [reply README.md:50]: should it?

File: README.md (hunk 3 above)
Comment (markpitchless) [reply README.md:15]: Ok, I give up! Where do you find this value? Can't see it in the Application setup any where in the UI.

File: README.md (hunk 4 above)
Comment (markpitchless) [reply README.md:56]: nope

**Pull request merged**